node/tests/SharedTests.ts

```diff
@@ -1233,6 +1233,184 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `hscan test_%p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key1 = "{key}-1" + uuidv4();
+                const key2 = "{key}-2" + uuidv4();
+                const initialCursor = "0";
+                const defaultCount = 20;
+                const resultCursorIndex = 0;
+                const resultCollectionIndex = 1;
+
+                // Setup test data - use a large number of entries to force an iterative cursor.
+                const numberMap: Record<string, string> = {};
+
+                for (let i = 0; i < 50000; i++) {
+                    numberMap[i.toString()] = "num" + i;
+                }
+
+                const charMembers = ["a", "b", "c", "d", "e"];
+                const charMap: Record<string, string> = {};
+
+                for (let i = 0; i < charMembers.length; i++) {
+                    charMap[charMembers[i]] = i.toString();
+                }
+
+                // Empty set
```

Comment:
nit: pull the empty set and negative cursor tests out into a separate test.  This will make this test have only a single responsibility.

Reply:
like hscan, sscan, zscan emptyset and negative cursor test like that?